src/Lachain.Core/Vault/KeyGenManager.cs

```diff
@@ -345,7 +346,7 @@ private TransactionReceipt MakeCommitTransaction(CommitMessage commitMessage, ul
                 commitMessage.Commitment.ToBytes(),
                 commitMessage.EncryptedRows
             );
-            return _transactionSigner.Sign(tx, _privateWallet.EcdsaKeyPair);
+            return _transactionSigner.Sign(tx, _privateWallet.EcdsaKeyPair, HardforkHeights.IsHardfork_6Active(_blockManager.GetHeight()));
```

Comment:
Should it be _blockManager.GetHeight() + 1 ?
or we can pass the blockHeight while calling.